index.js

```diff
@@ -56,32 +55,32 @@ app.get('/ideas', (req, res) => {
     let sortedIdeasList;
 
     if (req.query.sortBy === 'all') {
-        sortedIdeasList = getIdeas();
+        sortedIdeasList = getAllIdeas();
     } else if (req.query.sortBy === 'totalDonationSum') {
-        sortedIdeasList = sortedByDonationSumIdeas();
+        sortedIdeasList = sortIdeasByTotalDonationSum();
```

Comment:
The `sortIdeasByTotalDonationSum` function doesn't take any ideas as an argument here - so which ideas exactly it is going to sort? Can you improve this function?

Reply:
It returns the list with all ideas sorted in ascending order by totalDonationSum

Reply:
It's not clear from the name which ideas exactly it's sorting.

A clearer way to do it:
```js
const ideas = getIdeas();

const ideasSortedByTotalDonationSum = sortIdeasByTotalDonationSum(ideas);
```

When you pass `ideas` to `sortIdeasByTotalDonationSum` you're explicitly communicating which ideas you're sorting.

Reply:
I think I've understood, Updated. Along with  getPendingIdeas, getApprovedIdeas, getRejectedIdeas.

Reply:
` getPendingIdeas`, `getApprovedIdeas` and `getRejectedIdeas` is not the same as `sortIdeasByTotalDonationSum`.

`getPendingIdeas` name tells me that this function returns all pending ideas that you have - I don't need to provide anything to this function - it knows where to get all ideas, how to sort them and return.

`sortIdeasByTotalDonationSum` name tells me that this function is a utility function - I can provide _any_ list of ideas to it and it will sort them by total donation sum.

To conclude:

- `getPendingIdeas` - a use case-specific function, not applicable to any ideas
- `sortIdeasByTotalDonationSum` - a generic utility function, applicable to any ideas

Reply:
Oooooh, now I've understood. Fixed )